Update/v15 - New Timelines chapter (#295)
AvolitesLtd/TitanManual: docs/timelines/timeline-options.md

---
id: timeline-options
title: Timeline Options
sidebar_label: Timeline options
---

Timelines have a number of options which set how they operate. 
Press \<Options\> (or the \[Options\] softkey on the top level menu) then the **select** button for the timeline you wish to edit.

--- 

## Handle Tab

All options are the same as for a cue, see [Options](../cues/playback-options.md#handle-tab).

--- 

## Times Tab

![Timeline Options: Times](/docs/images/Timeline-Options-Times.png)

> These options let you set the end time and start time of the timeline. These times set the loop points when you are using the 

Reviewer: End time has been replaced by duration in beta 105

Author: Changed

more predictable when you are skipping around the timeline, otherwise you can get confusing LTP values
persisting from previous playbacks.

Option              | Action
--------------------|---
Override Playback Release | **Off**: The playback's own release settings are used.<br>On: The Timeline-Global release settings override the playback settings.
Release Playbacks to Home | Off: LTP attributes are left when when killing playbacks.<br>**On**: Attributes will return to home if no previous LTP values are set.
Timeline-Global Release Mask | Sets the Global Release Mask for playbacks triggered by the timeline (default **no attributes released**).
Timeline-Global Release Time | Sets the Global Release Time for playbacks triggered by the timeline (default **2 seconds**).

--- 

## Timecode Tab

![Timeline Options: Timecode](/docs/images/Timeline-Options-Timecode.png)

Option              | Action
--------------------|---
Autoload In Range | Off: Timeline must be fired manually. <br>**On:** Timeline will be fired automatically when timecode comes within the Start/End time range.

> Kill At Zero | **Off:** Timeline will remain active when the timeline fader is zeroed. <br>On: Timeline will be killed when the timeline fader is zeroed. 

Reviewer: I think we moved these options around

Author: These are the options as in beta 105

Reviewer 2: I think Fraser means moving the options between different groups - happened in #87419 build 15.0.114

---
id: timeline-options
title: Timeline Options
sidebar_label: Timeline options
---

Timelines have a number of options which set how they operate. 
Press \<Options\> (or the \[Options\] softkey on the top level menu) then the **select** button for the timeline you wish to edit.

--- 

## Handle Tab

> All options are the same as for a cue, see [Options](../cues/playback-options.md#handle-tab).

Reviewer 2: ```suggestion
All options are the same as for a memory, see [Options](../cues/playback-options.md#handle-tab).
```

Author: That is an interesting one. Everywhere else in the manual a single-state look is called a "cue". And the Options screen calls them "cue 20" (from the default legend which is Cue n). But if you press <Record> then the first option is [Create Memory]. So what is the correct term to use?

Reviewer 2: I agree it's an ongoing source of frustration, and there isn't a satisfactory solution without us removing some of the inconsistency in the UI (the default legend for a memory would be a great place to start, filed #87511). The problem with referring to a memory *as* a cue is that it's then ambiguous as to what the word "cue" means - does it mean the thing that lives within a chase/cuelist, or a memory (a "single-cue playback")? Within the software, a cue is a completely separate type of object to a playback - cues exist strictly *within* playbacks, and they have different properties, for example.

My general feeling is that we should settle on playbacks being one of three types; a memory, a cuelist or a chase. Cuelists and chases contain more than one cue, whereas "memory" is sort of a shorthand for a "single-cue playback". That might be out of the scope of this PR though if we currently use Cue everywhere! Don't know what others think (@AvolitesLtd/titan ?)

Author: From a user perspective, without internal knowledge of the software, I would say calling it a "cue" makes for easier understanding than calling it a "memory".  Then the term "cue list" makes sense as it appears to be a list of single cues (we try in the manual to refer to "steps" in a chase rather than "cues" to avoid that confusion). 

This is only an opinion from myself and a small range of Titan-using friends though, so your wider experience of Titan users may lead to a different answer.

Reviewer: Yeah agreed, I think there is some inconsistency in the software itself. I don't think we should worry about it for this PR and rather address the inconsistency more widely if we feel its needed.

Author: OK I will leave it saying "cue" here for now

--- 

> ## Times Tab

Reviewer 2: Maybe reference the context tool that can be used to crop the start/duration? #87423

![Timeline Options: Times](/docs/images/Timeline-Options-Times.png)

These options let you set the duration and start time of the timeline. These set the loop points when you are using the 
[Loop function](../timelines/timeline-options.md#timecode-tab) or are used to fire the timeline automatically when the timecode is in the correct range.
They can also be used to set limits on the timeline so you can be sure nothing will happen if you get sent some wayward timecode.

--- 

## Fader Tab

![Timeline Options: Fader](/docs/images/Timeline-Options-Fader.png)

Sets what happens to an internal timecode source when the playback fader containing the timeline is raised or zeroed. If an external
timecode source is being used then this option has no effect.

Option              | Action
--------------------|---
Fader Raised Action | **No Action**: The internal timecode source is not changed (default).<br>Play: The internal timecode source is started.
Fader Zero Action   | **No Action**: The internal timecode source is not changed (default).<br>Stop: The internal timecode source is stopped. <br>Pause: The internal timecode source is paused at the current time.

> > This option is useful for creating a complicated one-shot effect for busking. Create the effect as a timeline using internal timecode and set these options to **Play**/**Stop** - you can fire your effect just by raising the fader and when you lower the fader to zero the effect will reset.

Reviewer 2: Needs to be unlinked for a oneshot to work properly too

--- 

> ## Release Tab

Reviewer 2: Maybe worth clarifying as per other comment on this that `Timeline-Global Release` replaces your user's `Global Release` settings when things are done by the timeline.

![Timeline Options: Release](/docs/images/Timeline-Options-Release.png)

Sets what happens to LTP attributes in playbacks which are killed by the timeline. This can be useful to make the output
more predictable when you are skipping around the timeline, otherwise you can get confusing LTP values
persisting from previous playbacks.

Option              | Action
--------------------|---

> Override Playback Release | **Off**: The playback's own release settings are used.<br>On: The Timeline-Global release settings override the playback settings.
> Release Playbacks to Home | Off: LTP attributes are left when when killing playbacks.<br>**On**: Attributes will return to home if no previous LTP values are set.

Reviewer 2: ```suggestion
Override Playback Release | **Off**: The playback's own release settings are used (default).<br>On: The Timeline-Global release settings override the playback settings.
Release Playbacks to Home | Off: LTP attributes are left when when releasing the last playback controlling an attribute. <br>**On**: Attributes will return to home if no previous LTP values are set (default).
```

Reviewer 2: Also double `when`

Author: Default settings are indicated by the bold **Off** or **On**, I'm not putting in (default) every time

Reviewer 2: It was more for consitency - sometimes there was `(default)` (see Fader Raised / Zero). Don't really mind either way tbh

Author: Ah I had not noticed that... there is a mixture, unfortunately, as we move from having option lists in a big paragraph of text to the clearer way of showing them in a table, I am hoping to use the standard of showing default option as bold and not needing to say "default".

Timeline-Global Release Mask | Sets the Global Release Mask for playbacks triggered by the timeline (default **no attributes released**).
Timeline-Global Release Time | Sets the Global Release Time for playbacks triggered by the timeline (default **2 seconds**).

--- 

## Timecode Tab

![Timeline Options: Timecode](/docs/images/Timeline-Options-Timecode.png)

Option              | Action
--------------------|---
Autoload In Range | Off: Timeline must be fired manually. <br>**On:** Timeline will be fired automatically when timecode comes within the Start/End time range.
Kill At Zero | **Off:** Timeline will remain active when the timeline fader is zeroed. <br>On: Timeline will be killed when the timeline fader is zeroed. 

> Kill Out Of Range | Off: Timeline will remain active when the timecode goes outside the Start/End time range. <br>**On:** Timeline will be killed when the timecode goes outside the Start/End time range.
> Loop | **Off**: Internal timecode source continues to run<br>On: Internal timecode source will loop back to Start Time when it reaches End Time.

Reviewer 2: ```suggestion
Kill Out Of Range | Off: Timeline will remain active when the timecode goes outside the Start/End time range. <br>**On:** Timeline will be killed when the timecode goes outside the Start/End time range (default).
Loop | **Off**: Internal timecode source does not loop when it reaches End Time (default).<br>On: Internal timecode source will loop back to Start Time when it reaches End Time.
```
